opm/simulators/utils/HDF5File.cpp

```diff
@@ -123,4 +123,23 @@ void HDF5File::read(const std::string& group,
     H5Dclose(dataset_id);
 }
 
+int HDF5File::lastGroup() const
```

Comment:
1) nothing bad, atoi would return -1 and it won't be > 0 so it's just a nop.  
still, i have revised this to add a generic listing function to the hdf5file class, so we can keep the specifics in the serializer class. also i've added some sanity stuff on the serializer side.
furthermore let's stick the data as /report_step/n/xxx to minimize the chance of this happening.

2) yeah, added

3) that's all available in the hdf5 c++ bindings if we really need it, no reason for us to implement it. i've stuck to the C api for so long since the c++ api was troublesome back in 2004 when I started using HDF5. I assume it's fine now, but old habits and all that jazz..

4) lambda's (and std::function) are designed to be 100% compatible with C function callbacks. the compiler does all the magic.

Reply:
> lambda's (and `std::function`) are designed to be 100% compatible with C function callbacks. the compiler does all the magic.

Only if there are no captures/internal state.  If there is internal state in the lambda, then converting to a C-style function pointer is undefined behaviour.

Reply:
correct, i wasn't entirely precise.